fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java

```diff
@@ -74,61 +76,45 @@ public class EsExternalCatalog extends ExternalCatalog {
     /**
      * Default constructor for EsExternalCatalog.
      */
-    public EsExternalCatalog(long catalogId, String name, Map<String, String> props) throws DdlException {
+    public EsExternalCatalog(long catalogId, String name, Map<String, String> props) {
         this.id = catalogId;
         this.name = name;
         this.type = "es";
-        validate(props);
+        setProperties(props);
         this.catalogProperty = new CatalogProperty();
         this.catalogProperty.setProperties(props);
     }
 
-    private void validate(Map<String, String> properties) throws DdlException {
-        if (properties == null) {
-            throw new DdlException(
-                    "Please set properties of elasticsearch table, " + "they are: hosts, user, password, index");
-        }
-
-        if (StringUtils.isBlank(properties.get(PROP_HOSTS))) {
-            throw new DdlException("Hosts of ES table is null.");
-        }
-        nodes = properties.get(PROP_HOSTS).trim().split(",");
-        // check protocol
-        for (String seed : nodes) {
-            if (!seed.startsWith("http")) {
-                throw new DdlException("the protocol must be used");
-            }
+    private void setProperties(Map<String, String> properties) {
```

Comment:
I think we need to call `setProperties` in `gsonPostProcess` as well to make sure all properties are set after metadata replay.

Reply:
Fixed